### ISSUE TITLE
Warning: a promise was rejected with a non-error: [object Object]

### ISSUE THREAD

**Issue by QHose** [qlik-oss/enigma.js#163]
If I try to make a dataConnection, and I miss parameters (which is a bug in itself maybe), NodeJs reports `Warning: a promise was rejected with a non-error: [object Object]`

Is this expected?

```
I20170730-08:12:23.944(2)? ############### Connected
I20170730-08:12:23.947(2)? folder is  { qConnection:
I20170730-08:12:23.947(2)?    { qId: '',
I20170730-08:12:23.948(2)?      qName: 'dfd',
I20170730-08:12:23.949(2)?      qConnectionString: 'c:\\',
I20170730-08:12:23.950(2)?      qType: 'folder',
I20170730-08:12:23.951(2)?      qUserName: 'a',
I20170730-08:12:23.952(2)?      qPassword: 'a',
I20170730-08:12:23.956(2)?      qMeta: { qName: 'a' },
I20170730-08:12:23.957(2)?      qLogOn: 0 } }
W20170730-08:12:24.048(2)? (STDERR) Warning: a promise was rejected with a non-error: [object Object]
W20170730-08:12:24.049(2)? (STDERR)     at Session.processErrorInterceptor (C:\Users\Qlikexternal\Documents\GitHub\QRSMeteor\node_modules\enigma.js\dist\enigma.js:1936:30)
W20170730-08:12:24.050(2)? (STDERR)     at processImmediate [as _immediateCallback] (timers.js:396:17)
W20170730-08:12:24.051(2)? (STDERR) From previous event:
W20170730-08:12:24.052(2)? (STDERR)     at C:\Users\Qlikexternal\Documents\GitHub\QRSMeteor\node_modules\enigma.js\dist\enigma.js:1904:30
W20170730-08:12:24.053(2)? (STDERR)     at Array.reduce (native)
W20170730-08:12:24.054(2)? (STDERR)     at Session.intercept (C:\Users\Qlikexternal\Documents\GitHub\QRSMeteor\node_modules\enigma.js\dist\enigma.js:1903:28)
W20170730-08:12:24.055(2)? (STDERR)     at Session.send (C:\Users\Qlikexternal\Documents\GitHub\QRSMeteor\node_modules\enigma.js\dist\enigma.js:1577:27)
W20170730-08:12:24.057(2)? (STDERR)     at Object.fn [as createConnection] (C:\Users\Qlikexternal\Documents\GitHub\QRSMeteor\node_modules\enigma.js\dist\enigma.js:2776:32)
W20170730-08:12:24.061(2)? (STDERR)     at reloadAppAndReplaceScriptviaEngine$ (imports/api/server/QRSFunctionsApp.js:138:43)
W20170730-08:12:24.063(2)? (STDERR)     at tryCatch (C:\Users\Qlikexternal\Documents\GitHub\QRSMeteor\node_modules\regenerator-runtime\runtime.js:65:40)
W20170730-08:12:24.064(2)? (STDERR)     at GeneratorFunctionPrototype.invoke [as _invoke] (C:\Users\Qlikexternal\Documents\GitHub\QRSMeteor\node_modules\regenerator-runtime\runtime.js:303:22)
W20170730-08:12:24.065(2)? (STDERR)     at GeneratorFunctionPrototype.prototype.(anonymous function) [as next] (C:\Users\Qlikexternal\Documents\GitHub\QRSMeteor\node_modules\regenerator-runtime\runtime.js:117:21)
W20170730-08:12:24.067(2)? (STDERR)     at tryCatch (C:\Users\Qlikexternal\Documents\GitHub\QRSMeteor\node_modules\regenerator-runtime\runtime.js:65:40)
W20170730-08:12:24.070(2)? (STDERR)     at invoke (C:\Users\Qlikexternal\Documents\GitHub\QRSMeteor\node_modules\regenerator-runtime\runtime.js:155:20)
W20170730-08:12:24.071(2)? (STDERR)     at C:\Users\Qlikexternal\Documents\GitHub\QRSMeteor\node_modules\regenerator-runtime\runtime.js:165:13
W20170730-08:12:24.073(2)? (STDERR)     at C:\Users\Qlikexternal\AppData\Local\.meteor\packages\promise\0.8.9\npm\node_modules\meteor-promise\fiber_pool.js:32:39
W20170730-08:12:24.075(2)? (STDERR) error in reloadAppAndReplaceScriptviaEngine via Enigma.JS, did you used the correct schema definition in the settings.json file? { code: 9,
W20170730-08:12:24.076(2)? (STDERR)   parameter: 'Empty parameters',
W20170730-08:12:24.082(2)? (STDERR)   message: 'Empty parameters' }
```

```
 try {
        //connect to the engine
        var qix = await enigma.getService('qix', config);
        console.log('############### Connected');
        var call = {};
        call.action = 'Connect to Qlik Sense Engine API';
        call.request = 'Connect to Engine (using EnigmaJS) with a new appname parameter when you call global.openDoc: ', engineConfig.appname;
        call.url = gitHubLinks.replaceAndReloadApp;
        REST_Log(call, generationUserId);

        //create folder connection
        var folder = {
            // qConnection: {
            //     "qName": customer.name,
            //     "qType": "folder",
            //     "qConnectionString": customerDataFolder,
            //     "qLogOn": 0
            // }
            "qConnection": {
                "qId": "",
                "qName": "dfd",
                "qConnectionString": "c:\\",
                "qType": "folder",
                "qUserName": "a",
                "qPassword": "a",
                "qMeta": {
                    "qName": "a"
                },
                "qLogOn": 0
            }
        };
        console.log('folder is ', folder);
        var qConnectionId = await qix.app.createConnection(folder);
        console.log('created folder connection: ', qConnectionId);
```

**Comment by peol** [comment]
Hi @QHose. This is a warning logged by the bluebird promise library, not Node.js itself: http://bluebirdjs.com/docs/warning-explanations.html#warning-a-promise-was-rejected-with-a-non-error

While I partly agree with that definition (rejects should be `instanceof Error`), in this case we may lose information from the JSONRPC error (in this case the `missing params` error you see in your log).

I'll investigate if we can create error instances without breaking the API, but as said, it's a warning (not an error) from bluebird so your application should be running fine.

**Comment by peol** [comment]
Update:
AFAICT you will only get this warning if you have `NODE_ENV=development`, i.e. it will not be logged in a production environment. I'll continue investigating wrapping the errors in an instance though since it may provide valuable stack traces.

**Comment by QHose** [comment]
hi @peol , I am not an expert but should the promise not just be `rejected`?

**Comment by peol** [comment]
@QHose In your logs it says: `W20170730-08:12:24.048(2)? (STDERR) Warning: a promise was rejected with a non-error: [object Object]`, it is being rejected, but not with an expected `Error` instance (which should be fixed in v2.0.2).

**Comment by QHose** [comment]
haha, true! how could I miss that one :), good that it will be fixed!
